### DISCUSSION TITLE
Allow validation rules on create permissions to check relationship fields.

### DISCUSSION THREAD

**Discussion by u12206050** [directus/directus#23968] [Feature Requests]
### Summary

Allow validation logic specifically on create permissions to reference fields on relationships. Such that one can only create an item with reference to said relationship if those validation rules pass.

### Basic Example

You have `events` that people can register to by creating a items in the `registrations` collection.
Events have status `draft`, `published`, `open`, `closed`

People may view all non-draft events, but can only register (create a registration item) to events that have status `open` 
and can only `cancel` their registration (update the status of registration item) if the event status is `open | closed`

The permissions on `registrations` for create could have **validation** such as:
`{ event: { status: { _eq: 'open' } } }`

the update permissions already work and are:
`{ event: { status: { _in: ['open', 'closed'] } } }`

### Motivation

Since `create` permissions don't have any logic other than which fields can be set during creation, there is no way other than custom hooks to check that only valid relationships are being attached to this new item.

As in the example, when creating a new registration, the `event` field may only reference events that have status `open`.

### Detailed Design

The same logic checks already exist for checking permissions, so could also be used if any relationship fields are need to be validated.

### Must Have:
- A Relationship fields can be validated based on checks on the related item being attached.

### Should Have:
- B Since most relationships are being added via creation this would be the most important place to have it work.

### Could Have:
- D Filters in the UI could automatically apply on relationship fields where validation passes, meaning it could be different per role.

### Drawbacks

Slightly slower inserts due to the extra relational checking being done.

### Alternatives

1. Custom filter hook, but then isn't dynamic per role, unless implemented that way.
2. Attach everything you want to check to the $CURRENT_USER somehow eg. `open_events`, `closed_events` but this is terrible design.

### Adoption Strategy

New feature, no breaking change.

**Comment by JamesW1** [comment]
Thanks for the suggestion! We've moved feature requests over to our public roadmap at https://roadmap.directus.io/roadmap

I've logged this request for you, and if we see it raised multiple times, we'll consider adding it to the roadmap for prioritization.
